lib/SILOptimizer/Analysis/IsSelfRecursiveAnalysis.cpp

```diff
@@ -0,0 +1,47 @@
+//===--- IsSelfRecursiveAnalysis.cpp --------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/SILOptimizer/Analysis/IsSelfRecursiveAnalysis.h"
+#include "swift/SIL/ApplySite.h"
+#include "swift/SIL/SILBasicBlock.h"
+#include "swift/SIL/SILFunction.h"
+#include "swift/SIL/SILInstruction.h"
+
+using namespace swift;
+
+// Force the compiler to generate the destructor in this C++ file.
+// Otherwise it can happen that it is generated in a SwiftCompilerSources module
+// and that results in unresolved-symbols linker errors.
+IsSelfRecursive::~IsSelfRecursive() = default;
```

Comment:
It's not just the destructor right, isn't this the way to convince clang to place the vtable in this TU?